salt/thorium/reg.py

```diff
@@ -98,3 +98,31 @@ def mean(name, add, match):
             __reg__[name]['count'] += 1
             __reg__[name]['val'] = __reg__[name]['total'] / __reg__[name]['count']
     return ret
+
+
+def clear(name):
+    '''
+    Clear the register
+    '''
+    ret = {'name': name,
+           'changes': {},
+           'comment': '',
+           'result': True}
+    if name in __reg__ and isinstance(__reg__[name], set):
```

Comment:
This will be relatively slow. You shouldn't need to check the type nor instantiate a new object. Just `__reg__[name].clear()` will work on either one and it will be quick.